Core/src/org/sleuthkit/autopsy/logicalimager/dsp/AddMultipleImageTask.java

```diff
@@ -231,59 +266,64 @@ private void addImageToCase(String imageFilePath, List<Content> newDataSources,
                 errorMessages.add(Bundle.AddMultipleImageTask_criticalErrorAdding(imageFilePath, deviceId, ex.getLocalizedMessage()));
                 criticalErrorOccurred = true;
             }
-            /*
-             * Either way, the add image process needs to be reverted.
-             */
-            try {
-                addImageProcess.revert();
-            } catch (TskCoreException e) {
-                errorMessages.add(Bundle.AddMultipleImageTask_criticalErrorReverting(imageFilePath, deviceId, e.getLocalizedMessage()));
-                criticalErrorOccurred = true;
-            }
-            return;
         } catch (TskDataException ex) {
             errorMessages.add(Bundle.AddMultipleImageTask_nonCriticalErrorAdding(imageFilePath, deviceId, ex.getLocalizedMessage()));
         }
+    }
+    
+    /**
+     * Commits or reverts the results of the TSK add image process. If the
+     * process was stopped before it completed or there was a critical error the
+     * results are reverted, otherwise they are committed.
+     *
+     * @param imageFilePath  The image file path.
+     * @param errorMessages  Error messages, if any, are added to this list for
+     *                       eventual return via the callback.
+     * @param newDataSources If the new image is successfully committed, it is
+     *                       added to this list for eventual return via the
+     *                       callback.
+     */
+    private void commitOrRevertAddImageProcess(String imageFilePath, List<String> errorMessages, List<Content> newDataSources) {
+        synchronized (tskAddImageProcessLock) {
+            if (tskAddImageProcessStopped || criticalErrorOccurred) {
+                try {
+                    addImageProcess.revert();
+                } catch (TskCoreException ex) {
+                    errorMessages.add(Bundle.AddMultipleImageTask_criticalErrorReverting(imageFilePath, deviceId, ex.getLocalizedMessage()));
+                    criticalErrorOccurred = true;
+                }
+                return;
+            }
+        
+            if (!tskAddImageProcessStopped) {
```

Comment:
This if is unnecessary - it will have already returned if tskAddImageProcessStopped is true.